packages/diagrams/frontend/sirius-components-diagrams/src/dialog/useDialog.types.ts

```diff
@@ -0,0 +1,22 @@
+/*******************************************************************************
+ * Copyright (c) 2024 Obeo and others.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+import { GQLToolVariable } from '../renderer/palette/Palette.types';
+export interface UseDialogValue {
+  showDialog: (
+    dialogTypeId: string,
+    editingContextID: string,
+    dialogDescriptionId: string,
+    targetObjectId: string,
+    onConfirm: (variables: GQLToolVariable[]) => void
+  ) => void;
```

Comment:
I really don't understand this API.

- Why both a `diagramTypeId` and a `dialogDescriptionId` which are not even the same name as the GraphQL API
- `editingContextID` is not written this way, is that really necessary?
- Why give the targetObjectId?

Reply:
as discussed in the comments above, I am ok to remove the dialogType information if it can be owned by the dialogDescriptionID.
The TargetObjectId is needed by the selection dialog

Reply:
> The TargetObjectId is needed by the selection dialog

And the next dialog will ask for something else, we won't change the API anytime a dialog needs something. The API should not depend on the specific needs of the selection dialog.

Why can't the selection dialog get the currently selected node from a react flow hook or the current workbench selection from our `useSelection()` hook?

Reply:
> Why can't the selection dialog get the currently selected node from a react flow hook or the current workbench selection from our `useSelection()` hook?

I'm ok to use the hook. Are we sure that the selection cannot be changed in the meantime? (I don't see any particular scenario)

Reply:
done